### ISSUE TITLE
/markets/structures/{structure_id}/  should support type_id parameter

### ISSUE THREAD

**Issue by baronrabban** [esi/esi-issues#1255]
# Feature Request

/markets/structures/{structure_id}/  should support type_id parameter

**type_id**:  Return orders only for this type

## Use case

It's very inefficient to pull dozens of pages when I just want to know about a single type_id

## Authentication

The route is already authenticated.

## Example return

The same exact output but less.

**Comment by GoldenGnu** [comment]
Generally ESI does not do filtering, as it mess with caching. This endpoint is also authenticated and have to handle ACL, meaning it's even more expensive. If you add typeID filter, people may make requests for each type ID they need, instead of just getting everything.
In short: This would be expensive for the EVE severs, so the burden of filtering is put on us,  the users of the ESI.
Note: I'm just another user, I just wanted to explain to you why this will probably never happen.

**Comment by guiguilechat** [comment]
What he forgot to mention, is that parameter is already present on the regional market path.

**Comment by GoldenGnu** [comment]
The market region endpoint is not authed or and does not handle ACL, It's cached on AWS (or whatever cloud service ESI use ATM.) and not the internal ESI cache. Making the filter really really cheap, the market structures endpoint is not comparable to that.

Also, I'm not saying this shouldn't be done, just explain the design ESI usually follows and why this is very unlikely to be done.

**Comment by Jobeo** [comment]
Presently the TTT in Perimeter has 73 pages of results for 16,398,832 bytes of JSON.  If I want to look up the price of a mining laser there I have to iterate all 73 pages.  If I want to monitor that price all day the cost is 4.7GB downloaded data

Given https://github.com/esi/esi-issues/issues/704  there is no other way to see station range orders because the regional endpoint doesn't return them.

**Comment by guiguilechat** [comment]
Indeed, market region do not require to check access before returning the cache, therefore they can be cached further from the ESI, while structure market path being authed, the cache-control:private should prevent any intermediate caching.

**Comment by guiguilechat** [comment]
> If I want to look up the price of a mining laser there I have to iterate all 73 pages. If I want to monitor that price all day the cost is 4.7GB downloaded data

Indeed. So you should consider this cost before taking such an action. If you have a limited connection, then you should not host ESI-relying services : install your program on another server with no such a connection issue.
